http/org_service.go

```diff
@@ -426,6 +430,85 @@ func newOrganizationLogResponse(id influxdb.ID, es []*influxdb.OperationLogEntry
 	}
 }
 
+// SecretService connects to Influx via HTTP using tokens to manage secrets.
+type SecretService struct {
+	Client *httpc.Client
+}
+
+// LoadSecret is not implemented for http
+func (s *SecretService) LoadSecret(ctx context.Context, orgID influxdb.ID, k string) (string, error) {
+	return "", &influxdb.Error{
+		Code: influxdb.EMethodNotAllowed,
+		Msg:  "load secret is not implemented for http",
+	}
+}
+
+// PutSecret is not implemented for http.
+func (s *SecretService) PutSecret(ctx context.Context, orgID influxdb.ID, k string, v string) error {
+	return &influxdb.Error{
+		Code: influxdb.EMethodNotAllowed,
+		Msg:  "put secret is not implemented for http",
+	}
+}
+
+// GetSecretKeys get all secret keys mathing an org ID via HTTP.
+func (s *SecretService) GetSecretKeys(ctx context.Context, orgID influxdb.ID) ([]string, error) {
+	span, _ := tracing.StartSpanFromContext(ctx)
+	defer span.Finish()
+
+	span.LogKV("org-id", orgID)
+
+	path := strings.Replace(organizationsIDSecretsPath, ":id", orgID.String(), 1)
+
+	var ss secretsResponse
+	err := s.Client.
+		Get(path).
+		DecodeJSON(&ss).
+		Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return ss.Secrets, nil
+}
+
+// PutSecrets is not implemented for http.
+func (s *SecretService) PutSecrets(ctx context.Context, orgID influxdb.ID, m map[string]string) error {
+	return &influxdb.Error{
+		Code: influxdb.EMethodNotAllowed,
+		Msg:  "put secrets is not implemented for http",
+	}
+}
+
+// PatchSecrets will update the existing secret with new via http.
+func (s *SecretService) PatchSecrets(ctx context.Context, orgID influxdb.ID, m map[string]string) error {
+	span, _ := tracing.StartSpanFromContext(ctx)
+	defer span.Finish()
+
+	if orgID != 0 {
+		span.LogKV("org-id", orgID)
```

Comment:
I'm curious, we use the span's in some places, and not i others... feels like a service middleware concern to me.  Just wrap the service with a trace mw and its donezo that way 🤷‍♂

Reply:
same here, not sure what's that for

Reply:
lets go on and drop them for the time being then 👍